apps/bilateral_grid/Makefile

```diff
@@ -1,22 +1,22 @@
 include ../support/Makefile.inc
 
-all: filter
+all: $(BIN)/filter
 
-bilateral_grid: bilateral_grid.cpp
-	$(CXX) $(CXXFLAGS) bilateral_grid.cpp -g $(LIB_HALIDE) -o bilateral_grid -lz $(LDFLAGS) \
-	$(LLVM_SHARED_LIBS)
+$(BIN)/bilateral_grid.a: bilateral_grid_generator.cpp
+	@-mkdir -p $(BIN)
+	$(HALIDE_SRC_PATH)/tools/gengen.sh -c $(CXX) -l $(LIB_HALIDE) -o $(BIN) -s $^ target=host
```

Comment:
How much of a win is gengen.sh compared to compiling against GenGen.cpp directly? I'm wondering if the extra layer of indirection is really helpful.